### DISCUSSION TITLE
Is there a way to run Granian programmatically instead of from bash shell?

### DISCUSSION THREAD

**Discussion by DavydLiu** [emmett-framework/granian#254] [Q&A] [answered]
Wondering if we do something like Uvicorn?

```
import uvicorn

async def app(scope, receive, send):
    ...

if __name__ == "__main__":
    uvicorn.run("main:app", port=5000, log_level="info")
```

**Comment by gi0baro** [comment, marked as answer]
Adapting your snippet:

```python
import granian

async def app(scope, receive, send):
    ...

granian.Granian("main:app", interface="asgi", port=5000, log_level="info").serve()
```

You can check [Granian class](https://github.com/emmett-framework/granian/blob/v1.2.0/granian/server.py#L65) for further needs.

**Reply by DavydLiu** [reply]
Thanks for reply, I wanted to point out that when setting the interface I have to import the `Interface` constant from Granian. Pure text didn't work.
